src/lib/tools/formatToolOutput.ts

```
@@ -63,22 +67,54 @@ export function formatToolOutput(toolName: string, result: ToolResult): string {
+        }
+      }
+      // Fallback to JSON for unknown formats
+      return JSON.stringify(output);
+    }
+
     case 'navigation_tool': {
+      if (typeof output === 'string') {
+        return output;
+      }
       // Output: { url: string, success: boolean } or similar
       const navUrl = output.url || 'Unknown URL';
```

ESLint

[error] 'navUrl' is assigned a value but never used.

(@typescript-eslint/no-unused-vars)